File: 240322/Carry 피하기 2/escaping-carry-2.py
import sys

n = int(input())
nums = list(map(int,sys.stdin.read().splitlines()))

def carry(a,b):
    while a and b:
        if a%10 + b%10 >= 10:
            return False
        a //= 10
        b //= 10
    return True

ans = -1
for i in range(n):
    for j in range(i+1,n):
        if carry(nums[i],nums[j]):
            temp = nums[i] + nums[j]
            for k in range(j+1,n):
                if carry(temp,nums[k]):
                    ans = max(ans,temp+nums[k])

print(ans)

File: 240404/Carry 피하기 2/escaping-carry-2.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

int main() {
    int N, nums[20], A, B, C, ans = -1, temp;
    bool carry;

    scanf("%d",&N);
    for(int i = 0; i < N; i++)
        scanf("%d",&nums[i]);
    
    for(int i = 0; i < N; i++) {
        for(int j = i+1; j < N; j++) {
            for(int k = j+1; k < N; k++) {
                A = nums[i];
                B = nums[j];
                C = nums[k];
                carry = true;
                while (A > 0 || B > 0 || C > 0) {
                    if (A % 10 + B % 10 + C % 10 >= 10) {
                        carry = false;
                        break;
                    }
                    A /= 10;
                    B /= 10;
                    C /= 10;
                }
                if (carry) {
                    temp = nums[i]+nums[j]+nums[k];
                    if (ans < temp)
                        ans = temp;
                }
            }
        }
    }

    printf("%d\n", ans);
    return 0;
}